mantidproject/mantid
# Reflectometry interface: Event mode handling now avoids loading workspaces already present in ADS.

In the Reflectometry interface, the `GenericDataProcessorPresenter` has facilities for being able to access input runs from the AnalysisDataService rather than reloading them if possible. This PR allows the `ReflDataProcessorPresenter` class (which handles processing event workspaces) to access runs from the ADS in a similar manner.

**To test:**

This should be tested on `Release` mode. Alternatively if using `Debug` mode, go to `LoadEventNexus.cpp` and replace the following lines at lines 289-292 with this:

```
if (temp < 0) {
    if (temp < m_min_id)
        m_min_id = temp;
    if (temp > m_max_id)
        m_max_id = temp;
}
```

There are a variety of test cases that must pass for both event and non-event processing, To start:
- Interface can be accessed from Interfaces->Reflectometry->ISIS Reflectometry
- Enter runs `38415` and `38417` in the first two rows of the first group with transmission run `38393`.

For non-event testing, do not insert any values in the `Event Handling` tab. Then try the following:
- Process `38415` - this should load the run without failure.
- Load `38415` with `LoadISISNexus` as `TOF_38415` and process - this should successfully use run from ADS.
- Load `38415` with `LoadEventNexus` as `TOF_38415` and process - this should **not** use run from ADS and loads run instead.

For event testing, insert some values in the `Event Handling` tab (e.g. custom slicing: `0,10,20`). Then try these:
- Process `38415` - this should load the run without failure.
- Load `38415` with `LoadISISNexus` as `TOF_38415` and process - this should **not** use run from ADS and loads run instead.
- Load `38415` with `LoadEventNexus` as `TOF_38415` without loading monitors and process - this should **not** use run from ADS.
- Load `38415` with `LoadEventNexus` as `TOF_38415` with monitors and process - this should successfully use run from ADS.
- Repeat steps (apart from event workspaces) with `38417` - this is not an event workspace unlike `38415`. Ensure it uses the ADS when a loaded workspace for it exists.

Fixes #18536.

*Does not need to be in the release notes.*

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
